packages/interactivity/src/directives.js

```diff
@@ -226,24 +230,29 @@ export default () => {
 				const result = evaluate( entry );
 				element.props[ attribute ] = result;
 
-				// This seems necessary because Preact doesn't change the attributes
-				// on the hydration, so we have to do it manually. It doesn't need
-				// deps because it only needs to do it the first time.
+				/*
+				 * This is necessary because Preact doesn't change the attributes on the
+				 * hydration, so we have to do it manually. It only needs to do it the
+				 * first time. After that, Preact will handle the changes.
+				 */
 				useInit( () => {
 					const el = element.ref.current;
 
-					// We set the value directly to the corresponding
-					// HTMLElement instance property excluding the following
-					// special cases.
-					// We follow Preact's logic: https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L110-L129
+					/*
+					 * We set the value directly to the corresponding HTMLElement instance
+					 * property excluding the following special cases. We follow Preact's
+					 * logic: https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L110-L129
+					 */
 					if (
 						attribute !== 'width' &&
 						attribute !== 'height' &&
 						attribute !== 'href' &&
 						attribute !== 'list' &&
 						attribute !== 'form' &&
-						// Default value in browsers is `-1` and an empty string is
-						// cast to `0` instead
+						/*
+						 * Default value in browsers is `-1` and an empty string is
+						 * cast to `0` instead.
+						 */
```

Comment:
at first I thought this applied to the remaining attributes, but I think it's limited to `tabindex`. if that's the case it could be nice having that in the comment.

```php
/*
 * The value for `tabindex` follows the parsing rules for an integer.
 * if that fails, or if the attribute isn't present, then the browsers
 * should "follow platform conventions to determine if the element
 * should be considered as a focusable area," practically meaning that
 * most elements get a default of `-1` (not focusable), but several
 * also get a default of `0` (focusable in order after all elements
 * with a positive `tabindex` value).
 *
 * @see https://html.spec.whatwg.org/#tabindex-value
 */
```

I'm not sure why this is relevant here, but the comment jumped out at me and led me to investigate further. is it needed at all? is there a confusion that a comment should clarify?

Reply:
That part was copy-pasted [from Preact codebase](https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L110-L129), so whoever copied it left the same comment 🙂